keras/engine/training.py

```diff
@@ -962,10 +962,21 @@ def _make_predict_function(self):
                                                name='predict_function',
                                                **kwargs)
 
+    def _check_num_samples(self, ins, batch_size=None, steps=None, steps_name='steps'):
+        if steps is not None:
+            num_samples = steps
```

Comment:
It should be pretty simple to do with the same methods just by checking if either `x` or `y` is a Keras tensor in `fit(x, y, ...)`, so long as `steps_per_epoch` and `epochs` from `fit_generator()` are added to `fit(x, y, ..., steps_per_epoch=None, epochs=None)`.

Unless you have another suggestion, I propose checking for Keras tensors because it is the simplest mechanism by which multiple platforms, not just TF, could eventually support the API.  This means wrapping TF tensors with `Input()`, as in #6928 or after the more atomic PR merge order outlined in https://github.com/fchollet/keras/pull/7072#issuecomment-311174255.